src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const uniqueValues: any[] = [];
+
+  for (const item of items) {
+    const value = item[key];
+
+    if (!uniqueValues.includes(value)) {
+      uniqueValues.push(value);
+    }
+  }
+
+  uniqueValues.forEach((value) => {
+    result[value] = items.filter((item: T) => item[key] === value);
```

Comment:
Filtering `items` inside a loop can be very inefficient if `items` is large, as it results in an O(n^2) operation. You can optimize this by iterating through `items` only once and grouping them as you go, without the need for the `uniqueValues` array or the `filter` operation.